namespaces/referencing-namespace.md

```diff
@@ -0,0 +1,194 @@
+
+It MAY also be used as an attribute if defined by the particular namespace or element, and its used depends on how it is defined to be used.
+
+#### Properties
+
+- `element`: namespace (string, required)
+- `attributes`
+    - `prefixAs` (string, optional) - Unique name to be used to reference this namespace throughout the document.
+- `content` (string, required) - Resolvable URI of namespace, which MAY be relative or absolute
+
+#### Example
+
+```json
+{
+  "element": "require",
+  "content": [
+    {
+      "element": "namespace",
+      "content": "http://example.com/foo"
+    },
+    {
+      "element": "namespace",
+      "attributes": {
+        "prefixAs": "bar"
+      },
+      "content": "http://example.com/bar"
+    }
+  ]
+}
+```
+
+There are two namespaces used here, one without the prefix and one with the prefix.
+
+- The first namespace defined for `http://example.com/foo` MUST have all of its elements available in the document in which it is required. If there is a conflict in element names, the namespace defined latest in the document MUST receive precedence.
+- The second namespace for `http://example.com/bar` has a prefix specified, which means its elements will MUST NOT be included in the document by default, but MUST be made available for referencing using the given prefix. If there is a conflict in prefix names, the prefix defined last MUST receive precedence.
+
+### Type (Element)
+
+The `type` element provides a way to add additional data about an element. In regards to how it is defined here, it is used for specifying a namespace in which the element may be found.
+
+#### Properties
+
+- `element`: type (string, required)
+- `attributes`
+   - `ref` (string) - Link directly to an element using the element's ID in the fragment identifier
+   - `namespace` (string) - Specify the namespace in which the element may be found
+   - `prefix` (string) - Specify the prefix of the required namespace in which the element may be found
+- `content` (string, required) - The name of the element
+
+#### Example
+
+```json
+{
+  "element": "items",
+  "content": [
+    {
+      "element": "foo",
+      "content": 1
+    },
+    {
+      "element": {
+        "element": "type",
+        "attributes": {
+          "ref": "http://example.com/bar#foo"
+        },
+        "content": "foo"
+      },
+      "content": 2
+    },
+    {
+      "element": {
+        "element": "type",
+        "attributes": {
+          "namespace": "http://example.com/bar"
+        },
+        "content": "foo"
+      },
+      "content": 3
+    },
+    {
+      "element": {
+        "element": "type",
+        "attributes": {
+          "prefix": "bar"
+        },
+        "content": "foo"
+      },
+      "content": 4
+    }
+  ]
+}
+```
+
+This example shows an `items` element that has four elements as its content.
```

Comment:
I was addressing concerns with putting things within the attributes area. Are you OK with having this all in attributes like this? It does reduce complexity.